tests/kernel/mem_barrier/CMakeLists.txt

```diff
@@ -0,0 +1,7 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
```

Comment:
Mild complaint here: a new test adds a few seconds of build time for every platform involved in CI.  You're configuring and building the whole OS again just for one tiny function of like six instructions.  Maybe add this as a case somewhere else?  tests/kernel/common would be a fine spot as it's never filtered.